prowler/providers/aws/services/stepfunctions/stepfunctions_service.py

```diff
@@ -0,0 +1,341 @@
+        include_execution_data (bool): Whether to include execution data in the logs.
+        destinations (List[LoggingDestination]): List of logging destinations.
+    """
+
+    level: LoggingLevel
+    include_execution_data: bool
+    destinations: List[LoggingDestination]
+
+
+class TracingConfiguration(BaseModel):
+    """
+    Represents the tracing configuration for a State Machine.
+
+    Attributes:
+        enabled (bool): Whether X-Ray tracing is enabled.
+    """
+
+    enabled: bool
+
+
+class EncryptionConfiguration(BaseModel):
+    """
+    Represents the encryption configuration for a State Machine.
+
+    Attributes:
+        kms_key_id (Optional[str]): The KMS key ID used for encryption.
+        kms_data_key_reuse_period_seconds (Optional[int]): The time in seconds that a KMS data key can be reused.
+        type (EncryptionType): The type of encryption used.
+    """
+
+    kms_key_id: Optional[str]
+    kms_data_key_reuse_period_seconds: Optional[int]
+    type: EncryptionType
+
+
+class StateMachine(BaseModel):
+    """
+    Represents an AWS Step Functions State Machine.
+
+    Attributes:
+        id (str): The unique identifier of the state machine.
+        arn (str): The ARN of the state machine.
+        name (Optional[str]): The name of the state machine.
+        status (StateMachineStatus): The current status of the state machine.
+        definition (str): The Amazon States Language definition of the state machine.
+        role_arn (str): The ARN of the IAM role used by the state machine.
+        type (StateMachineType): The type of the state machine (STANDARD or EXPRESS).
+        creation_date (datetime): The creation date and time of the state machine.
+        region (str): The region where the state machine is.
+        logging_configuration (Optional[LoggingConfiguration]): The logging configuration of the state machine.
+        tracing_configuration (Optional[TracingConfiguration]): The tracing configuration of the state machine.
+        label (Optional[str]): The label associated with the state machine.
+        revision_id (Optional[str]): The revision ID of the state machine.
+        description (Optional[str]): A description of the state machine.
+        encryption_configuration (Optional[EncryptionConfiguration]): The encryption configuration of the state machine.
+        tags (List[Dict]): A list of tags associated with the state machine.
+    """
+
+    id: str
+    arn: str
+    name: Optional[str] = None
+    status: StateMachineStatus
+    definition: str
+    role_arn: str
+    type: StateMachineType
+    creation_date: datetime
+    region: str
+    logging_configuration: Optional[LoggingConfiguration] = None
+    tracing_configuration: Optional[TracingConfiguration] = None
+    label: Optional[str] = None
+    revision_id: Optional[str] = None
+    description: Optional[str] = None
+    encryption_configuration: Optional[EncryptionConfiguration] = None
```

Comment:
Why all of these attributes if you only need the `logging_configuration` ?

Reply:
To make it easier for users who want to add checks that do not have to change the service, I decided to add as much information as necessary, as long as I do not have to make extra calls